src/app/compiler/compiler.js

```diff
@@ -219,10 +282,14 @@ function Compiler (handleImportCall) {
     worker.postMessage({cmd: 'loadVersion', data: url})
   }
 
-  function gatherImports (files, target, importHints, cb) {
+  function gatherImports (files, importHints, cb) {
     importHints = importHints || []
+    var fileList = Object.keys(files)
+    if (!fileList.length) {
+      return cb(null, {})
+    }
     if (!compilerAcceptsMultipleFiles) {
-      cb(null, files[target])
+      cb(null, files[fileList[0]])
```

Comment:
I am not sure `fileList[0]` will work in every situation? Have you tried multiple imports and import chains? With and without failing imports?

Reply:
That is the same behavior as before:  we just take one file, and as we don't have `target` anymore we take the first one.
I plan to add more test in the coming days, testing import more carefully is planned

Reply:
I'm not sure if `target` always equals to the first file.

Reply:
well actually I don't really get where `target` is used...
except here: https://github.com/ethereum/browser-solidity/blob/master/src/app/compiler/compiler.js#L239
but this seems wrong to me...
rather `var path = /(.*\/).*/.exec(fileName)` than `var path = /(.*\/).*/.exec(target)`

Reply:
I am not sure, it is one of those magic codes which worked at some point and if I recall correctly it was to avoid issues with imports (perhaps loops - where A imports B and B imports A). solc handles this properly, but remix does its own import feature before solc.